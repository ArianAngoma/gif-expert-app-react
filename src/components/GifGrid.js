import {useState, useEffect} from "react";
import GifGridItem from "./GifGridItem";

const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    // Hook para que la funcion getGif() solo se ejecute una vez a pesar de haber cambios en el estado del componente
    useEffect(() => {
        getGif();
    }, [])

    const getGif = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=gOrIoBdvHeKsTQqOu8XnLe60FI4Q2Yx5';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);
        setImages(gifs);
    }

    return (
        <div>
            <h3>{category}</h3>
            {
                images.map(img => (
                    <GifGridItem
                        key={img.id}
                        {...img}/>
                ))
            }
        </div>
    )
}

export default GifGrid;