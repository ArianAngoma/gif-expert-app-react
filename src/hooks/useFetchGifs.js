import {useEffect, useState} from "react";
import {getGifs} from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    // Hook para que la funcion getGif() solo se ejecute una vez a pesar de haber cambios en el estado del componente
    useEffect(() => {
        getGifs(category).then(images => {
            setState({
                data: images,
                loading: false
            })
        });
    }, [category]);

    return state; // {data: [], loading: true}
}