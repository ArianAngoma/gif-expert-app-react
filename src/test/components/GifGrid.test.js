import {shallow} from "enzyme";
import GifGrid from "../../components/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Test to component <GifGrid/>', () => {
    const category = 'Anime'


    test('Should display the component <GifGrid/> correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when loading images useFetchGifs', () => {
        useFetchGifs.mockReturnValue({
            data: [{id: 'ayMW3eqvuP00o', title: 'No Face GIF by Spirited Away', url: 'https://giphy.com/gifs/spirited-away-no-face-ayMW3eqvuP00o'}],
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });
})