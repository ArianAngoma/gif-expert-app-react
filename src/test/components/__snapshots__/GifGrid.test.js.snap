// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to component <GifGrid/> Should display the component <GifGrid/> correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Anime
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test to component <GifGrid/> should show items when loading images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Anime
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ayMW3eqvuP00o"
      key="ayMW3eqvuP00o"
      title="No Face GIF by Spirited Away"
      url="https://giphy.com/gifs/spirited-away-no-face-ayMW3eqvuP00o"
    />
  </div>
</Fragment>
`;
